CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -8,107 +8,88 @@
     {
         Map<String,Object> m = new LinkedHashMap<>();
         m.put("file", filename);
         m.put("segment", numPacket);
         m.put("lines", file.toArray());
         
-        
         oClient.sendNotification(m);
     }
-    static String setFieldName(int i , Map<String,String>schema)
+    static String setFieldName(int i, Map<String,String>schema)
     {
-        String field = String.format("field%d",i); 
-        if (schema != null)
-        {
-            if (schema.get(field)!= null)
-            {
+        String field = String.format("field%d", i); 
+        if (schema != null){
+            if (schema.get(field)!= null){
                 field = schema.get(field) ; 
             }
         }
-
         return field ; 
     }
-    @SuppressWarnings("unchecked")
-    static public ArrayList<Map<String,String>> execute(String csvFile,  Map<String, Object>     config,   ExtensionWebSocketClient oClient  ) {
 
+    @SuppressWarnings("unchecked")
+    static public ArrayList<Map<String,String>> execute(String csvFile, Map<String, Object> config, ExtensionWebSocketClient oClient) {
         String line = "";
         int numOfRecords ; 
-        int numOfProcessedLinesInSegment = 0 ; 
+        int numOfProcessedLinesFromFile = 0 ; 
         Map<String,String> schema = null;
 
-        if (config.get("schema") != null)
-        {
+        if (config.get("schema") != null){
             schema = (Map<String,String>) config.get("schema");
         }
 
-        String delimiter = ";";
-        if (config.get("delimiter") != null)
-        {
+        String delimiter = ",";
+        if (config.get("delimiter") != null){
             delimiter = config.get("delimiter").toString();
         }
 
-
         int numLinesInEvent = (int) config.get(MAX_LINES_IN_EVENT);
-
         ArrayList<Map<String,String>> file = new ArrayList<Map<String,String>>(); 
 
         try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
-
             numOfRecords = 0 ; 
             while ((line = br.readLine()) != null) {
-                numOfProcessedLinesInSegment++;
+                numOfProcessedLinesFromFile++;
                 // use comma as separator
                 String[] values = line.split(delimiter);
-
                 Map<String, String> lineValues = new HashMap<String, String>(); 
                 
-                for (int i = 0 ; i < values.length ; i++)
-                {
-                    String currField = setFieldName(i,schema);
-                    lineValues.put(currField, values[i]);
+                int schemaFieldIndex = 0 ; 
+                for (int i = 0 ; i < values.length ; i++){
+                    if (values[i].length() != 0){
+                        String currField = setFieldName(schemaFieldIndex,schema);
+                        lineValues.put(currField, values[i]);
+                        schemaFieldIndex++;
+                    }
```

Comment:
currently , null value doesn't increment the schema index field. so in the first example it would get field0=fred.
as i don't see it happen in real life,but  i will add additional configuration flag which will give control to the user to decide about the behavior.